server/server.js

```diff
@@ -1,56 +1,88 @@
 var express = require('express');
 var app = express();
 const db = require('./mysql/data.js');
+var bodyParser = require('body-parser');
 
-app.use(function(req, res, next) {
+/**
+ * body-parser 是必须要有的！ 不然request body是空的！
+ * Body-parser is the Node. js body parsing middleware.
+ * It is responsible for parsing the incoming request 
+ * bodies in a middleware before you handle it.
+ */
+
+app.use(function (req, res, next) {
     //add a header message to solve cors security policy access deny problem
     res.header("Access-Control-Allow-Origin", "http://localhost:3000"); // update to match the domain you will make the request from
     res.header("Access-Control-Allow-Headers", "Origin, X-Requested-With, Content-Type, Accept");
+    res.header("Access-Control-Allow-Methods", "POST, GET, DELETE, OPTIONS");
     next();
-  });
+});
+
+app.use(bodyParser.urlencoded({ extended: true }));
+app.use(bodyParser.json());
 
-app.get('/', async (req, res, next) =>{
-    console.log(req.body);
-    db.query('select * from master where contained_by = 0', [],function(result,fields){
+
+//home page's data
+app.get('/', async (req, res, next) => {
+    db.query('select * from master where contained_by = 0', [], function (result, fields) {
         console.log('查询结果：');
         console.log(result);
 
-        res.end( JSON.stringify(result));
+        res.end(JSON.stringify(result));
     });
 });
 
-app.get('/:id',function(req,res, next){
+//get specific item's data
+app.get('/:id', function (req, res, next) {
     var id = req.params.id;
-    db.query(`select * from master where contained_by = ${id}`, [],function(result,fields){
+    db.query(`select * from master where contained_by = ${id}`, [], function (result, fields) {
         console.log('查询结果：');
         console.log(result);
 
-        res.end( JSON.stringify(result));
+        res.end(JSON.stringify(result));
     });
 });
 
-app.post('/addItem', function (req, res) {
-    db.query('select * from master', [],function(result,fields){
-        var toAdd = req.body;
+//get current container info
+app.get('/containerInfo/:id', function (req, res, next) {
+    var id = req.params.id;
+    db.query(`select * from master where item_id = ${id}`, [], function (result, fields) {
         console.log('查询结果：');
         console.log(result);
 
-        res.end();
+        res.end(JSON.stringify(result));
     });
- });
+});
 
- app.post('/container', function (req, res) {
-    db.query('select * from master', [],function(result,fields){
-        var toAdd = req.body;
-        console.log('查询结果：');
+//add new item
+app.post('/addItem', function (req, res) {
```

Comment:
Please use `/item` url with post request
```suggestion
app.post('/item', function (req, res) {
```